packages/core/src/actions.ts

```diff
@@ -157,10 +158,14 @@ export function toActivityDefinition<TContext, TEvent extends EventObject>(
  *
  * @param eventType The event to raise.
  */
-export function raise<TContext, TEvent extends EventObject>(
-  event: Event<TEvent>,
-  options?: RaiseActionOptions<TContext, TEvent>
-): RaiseAction<TEvent> {
+export function raise<
+  TContext,
+  TExpressionEvent extends EventObject,
+  TEvent extends EventObject = TExpressionEvent
+>(
+  event: NoInfer<TEvent>,
```

Comment:
previously this was typed as `Event<TEvent>`:
https://github.com/statelyai/xstate/blob/70f426a024616be2f0a77bcfb9ad9a74c87844d7/packages/core/src/types.ts#L90

This wasn't correct for events with payload. IIRC we've decided to ditch the idea of extracting "simple" events and allowing them to be used as strings. I think it would be OK-ish to introduce such a "breaking change" (type-level only) in a minor version. Let me know what do you think.

Reply:
Yeah, that's fine. Worst-case scenario, it would only "break" code at the type-level (ignorable) but still work exactly the same.